### PR TITLE
core: static mapping of non-secure DDR and of secure memory with read-only rights

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: core/mm/core_mmu.c [hunk 1]
```diff
@@ -628,9 +628,18 @@ static void verify_special_mem_areas(struct tee_mmap_region *mem_map,
...
+	 *
+	 * Exceptions are the memory areas that maps with the same attributes
+	 * as for example MEM_AREA_RAM_NSEC and MEM_AREA_NSEC_SHM
+	 * which may overlap since they are used for the same purpose
+	 * except that MEM_AREA_NSEC_SHM is always mapped and
+	 * MEM_AREA_RAM_NSEC only uses a dynamic mapping.
 	 */
 	for (mem = start; mem < end; mem++) {
 		for (mmap = mem_map; mmap->type != MEM_AREA_END; mmap++) {
+			if (core_mmu_type_to_attr(mem->type) ==
```
Comment (jenswi-linaro) [inline core/mm/core_mmu.c:641]: I'd rather be explicit about which memory area we allow to overlap. The purpose here is to check against unintended overlapping, so let's make it easy to keep track of the allowed exceptions.
Currently, it seems it's only with `MEM_AREA_RAM_NSEC` and `MEM_AREA_NSEC_SHM` where it makes sense to allow an overlap.

Comment (etienne-lms) [reply core/mm/core_mmu.c:641]: fair. i'll update accordingly.

Comment (etienne-lms) [reply core/mm/core_mmu.c]: I've remove that ch

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

Nit: Should the SoBs in the last commit be swapped?

**Comment by etienne-lms** [comment]
I've swapped theS-o-b tags in commit "core: mmu: fix nsec ddr discovery regarding MEM_AREA_RAM_NSEC" and applied the review tag to the 3 commits.
Thanks for the review.

**Pull request merged**